lib_eio/stream.mli

```diff
@@ -40,6 +40,15 @@ val take_nonblocking : 'a t -> 'a option
     Note that if another domain may add to the stream then a [None]
     result may already be out-of-date by the time this returns. *)
 
+val select: ('a t * ('a -> 'b)) list -> 'b
+(** [select] waits for any stream to have an item available. The item
+    is mapped by the provided function and returned. Example:
+    [select [(s1, fun x -> x+1); (s2, fun x -> x+2)]
+
+    Warning: as with `Fiber.first`, it is possible that two or more streams
+    yield an item simultaneously, in which case only one item will be
+    returned, and the other items are discarded.*)
```

Comment:
In my opinion, this caveat almost defeats the point of the function.
I would love to see `Eio.Stream.select` added, but I think it needs to be done using something like #558